lib/modules/hosts/youtube.js

```diff
@@ -5,7 +5,7 @@ export default {
 	name: 'youtube',
 	attribution: false,
 	domains: ['youtube.com', 'youtu.be'],
-	detect: ({ href }) => (/^https?:\/\/(?:youtu\.be|(?:www\.|m\.)?youtube\.com)\/?(?:watch|embed)?(?:.*?v=|v\/|\/)([\w-]+)/i).exec(href),
+	detect: ({ href }) => (/^https?:\/\/(?:www\.|m\.)?(?:youtu\.be\/|youtube\.com\/?(?:watch|embed)(?:.*?v=|v\/|\/))([\w-]+)/i).exec(href),
```

Comment:
I think I'd change it a bit more, but that looks good:

```js
detect: ({ pathname, hostname, search }) => {
    const params = new URLSearchParams(search);
    const split = pathname.substring(1).split('/');
    // short url
    if (hostname.endsWith('youtu.be')) return split[0];
    // long url
    return params.has('v') ?
        params.get('v') : // ?v=
        ['watch', 'embed', 'v'].includes(split[0]) && split[1]; // watch/embed/v
},
```

And you'd have to change the handleLink param from `[, hash]` to `hash`

I removed `videoIdRe` because unless the URL is _really_ messed up, it'll just match everything. But I don't feel strongly about this, you could put it back if you want.